File: Java/Arrays/1200 MinimumAbsoluteDifference.java
// LeetCode 1200. Minimum Absolute Difference
// Time Complexity: O(n log n)
// Space Complexity: O(1) (excluding output)

import java.util.*;

class Solution {
    public List<List<Integer>> minimumAbsDifference(int[] arr) {
        Arrays.sort(arr);
        List<List<Integer>> res = new ArrayList<>();

        int n = arr.length;
        int diff = Integer.MAX_VALUE;

        // Find minimum absolute difference
        for (int i = 1; i < n; i++) {
            diff = Math.min(diff, arr[i] - arr[i - 1]);
        }

        // Collect all pairs with that difference
        for (int i = 1; i < n; i++) {
            if (arr[i] - arr[i - 1] == diff) {
                res.add(List.of(arr[i - 1], arr[i]));
            }
        }

        return res;
    }
}

File: C++/Arrays/1200 MinimumAbsoluteDifference.cpp
// LeetCode 1200. Minimum Absolute Difference
// Time Complexity: O(n log n)
// Space Complexity: O(1) (excluding output)

#include <vector>
#include <algorithm>
using namespace std;

class Solution {
public:
    vector<vector<int>> minimumAbsDifference(vector<int>& arr) {
        sort(arr.begin(), arr.end());
        vector<vector<int>> res;

        int n = arr.size();
        int diff = INT_MAX;

        // Find minimum absolute difference
        for (int i = 1; i < n; i++) {
            diff = min(diff, arr[i] - arr[i - 1]);
        }

        // Collect all valid pairs
        for (int i = 1; i < n; i++) {
            if (arr[i] - arr[i - 1] == diff) {
                res.push_back({arr[i - 1], arr[i]});
            }
        }

        return res;
    }
};
